### DISCUSSION TITLE
Long Time for IMU Self-Calibration

### DISCUSSION THREAD

**Discussion by lyc-king** [bitcraze/discussions#365] [Q&A]
Dear all,

I met a problem about the long time of IMU self-calibration on the Crazyflie. Every time I restart it, I have to wait for its initialization for a long time and sometimes it just doesn't work with red light flashing slowly.

Did anyone meet similar problem? How did you solve that, please? Thanks a lot for any suggestions!

**Comment by tobbeanton** [comment]
There is a [topic](https://forum.bitcraze.io/viewtopic.php?p=23219&hilit=20000#p23219) about this in our old forum. The variance calculation has been corrected since then and a value of ~150 is probably good to try with now. The BIAS define is [here](https://github.com/bitcraze/crazyflie-firmware/blob/d7786addc03c54819fc78138198b6633b57b7104/src/hal/src/sensors_bmi088_bmp388.c#L86).
